service/src/main/java/com/codeforcommunity/processor/ImportProcessorImpl.java

```diff
@@ -0,0 +1,59 @@
+package com.codeforcommunity.processor;
+
+import com.codeforcommunity.api.IImportProcessor;
+import com.codeforcommunity.auth.JWTData;
+import com.codeforcommunity.dto.imports.BlockImport;
+import com.codeforcommunity.dto.imports.ImportBlocksRequest;
+import com.codeforcommunity.dto.imports.ImportNeighborhoodsRequest;
+import com.codeforcommunity.dto.imports.NeighborhoodImport;
+import com.codeforcommunity.enums.PrivilegeLevel;
+import com.codeforcommunity.exceptions.AuthException;
+import org.jooq.DSLContext;
+import org.jooq.generated.Tables;
+import org.jooq.generated.tables.records.BlocksRecord;
+import org.jooq.generated.tables.records.NeighborhoodsRecord;
+
+public class ImportProcessorImpl implements IImportProcessor {
+  private final DSLContext db;
+
+  public ImportProcessorImpl(DSLContext db) {
+    this.db = db;
+  }
+
+  @Override
+  public void importBlocks(JWTData userData, ImportBlocksRequest importBlocksRequest) {
```

Comment:
Should there be any checks that there is data in the neighborhoods table before being able to run this route? Since blocks are foreign keyed to neighborhoods storing them without an existing neighborhood would throw an error I believe.

Reply:
That's true. The reason I didn't originally is because it's kind of unavoidable that we're exposing ourselves to foreign key exceptions unless we check each ID before we insert it. Having a baseline check that just checks that there's _something_ there could be helpful though, what do you think?

Reply:
I think a check for having something there is good and easy to implement. Checking each id seems like too much and a whole different issue if that is not adding up versus whether someone forgot to call the importNeighborhoods route before.